stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -385,6 +363,68 @@ def _get_rule_from_delivery_route(self, html_exc=False):
             )
         return rule
 
+    def _is_valid_for_move(self, move):
+        self.ensure_one()
+        if not self.move_domain:
+            return move
+        domain = safe_eval(self.move_domain or "[]")
+        if not domain:
+            return move
+        return move.filtered_domain(domain)
+
+    def _prepare_move_split_vals(self, move, split_qty):
+        split_qty_uom = move.product_id.uom_id._compute_quantity(
+            split_qty, move.product_uom, rounding_method="HALF-UP"
+        )
+        return move._prepare_move_split_vals(split_qty_uom)
+
+    def _split_move(self, move, split_qty):
+        split_move = move.copy(self._prepare_move_split_vals(move, split_qty))
+        new_product_qty = move.product_id.uom_id._compute_quantity(
+            move.product_qty - split_qty, move.product_uom, round=False
+        )
+        move.product_uom_qty = new_product_qty
+        return split_move
+
+    def _compute_split_qty_by_packaging(self, move, packaging):
+        """Returns the qty to split for move and packaging"""
+        rounding = move.product_id.uom_id.rounding
+        if (
+            packaging.packaging_type_id not in self.packaging_type_ids
```

Comment:
I would move this test `if packaging.packaging_type_id not in self.packaging_type_ids`